src/core/infer_request.cc

```diff
@@ -562,6 +562,62 @@ InferenceRequest::Normalize()
   return Status::Success;
 }
 
+void
+InferenceRequest::Report(
+    bool success, int device, uint64_t compute_start_ns,
+    uint64_t compute_input_end_ns, uint64_t compute_output_start_ns,
+    uint64_t compute_end_ns, uint64_t compute_input_duration_ns,
+    uint64_t compute_infer_duration_ns, uint64_t compute_output_duration_ns)
+{
+  struct timespec last_ts;
+
+  clock_gettime(CLOCK_REALTIME, &last_ts);
+  const uint64_t last_timestamp_ms = TIMESPEC_TO_MILLIS(last_ts);
+
+  // Need a monotonic clock for request end time
+  clock_gettime(CLOCK_MONOTONIC, &last_ts);
+  auto request_duration_ns = TIMESPEC_TO_NANOS(last_ts) - request_start_ns_;
+
+  if (success) {
```

Comment:
In these calls to the aggregator just pass in the timestamps and have the aggregator compute the deltas.